src/smt.rs

```diff
@@ -68,6 +68,7 @@ pub enum SMTOp {
     And,
     Or,
     Ite,
+    Impl,
```

Comment:
Can you name it `Implies` instead? Lower it looks confusing to me vs Rust's Impl